.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,52 @@
+name: Run e2e tests on Kedro-Viz
+# Runs end-to-end tests on Kedro-Viz for different 
+# operating systems and Python versions.
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        type: string
+      python-version:
+        type: string
+jobs:
+    e2e_tests:
+        runs-on: ${{ inputs.os }}
+        # Only run on main and demo branches for Windows
+        if: inputs.os == 'ubuntu-latest' || ((github.ref == 'refs/heads/main' || github.ref == 'refs/heads/demo') && inputs.os == 'windows-latest')
```

Comment:
Just out of curiosity, is there a reason it was set up like this on CircleCI?

Reply:
@ankatiyar  Because testing on Windows is time-consuming, we have decided for simplicity to only test on the main and demo versions. Therefore, for each commit, we do not perform full testing; instead, we conduct it when merging to the main and demo branches.

Reply:
Makes sense!

Reply:
@ravi-kumar-pilla this condition ensures that the tests will run on Ubuntu regardless of the branch, and for Windows, the tests will only run if the branch is either "main" or "demo".  So for PRs to main this will works regardless of the OS.

Reply:
Yes, #1853 has issue on windows. So the tests failed after merging to main. So, I was referring to running the tests on windows (atleast 1 python version) to be safe when a PR is raised to main

Reply:
@ravi-kumar-pilla Okay got it.
This workflow(e2e-tests.yml) and parent workflow (build-backend.yml) will be trigger when a PR is raised to main for windows as well.

Reply:
@jitu5 , @ravi-kumar-pilla  I suggest we stick to our previous approach. Whether we run one Windows test or multiple tests in parallel, they take approximately the same amount of time, around 30 minutes. This duration is double that of running Python tests. Waiting for 30 minutes for tests to complete after pushing to a PR wasn't a great dev experience, hence this change was introduced. Windows errors are usually rare, so it's ok to address them reactively once the changes are merged into the main branch.